src/dbnode/persist/fs/write.go

```diff
@@ -327,13 +329,38 @@ func (w *writer) Close() error {
 	}
 	// NB(xichen): only write out the checkpoint file if there are no errors
 	// encountered between calling writer.Open() and writer.Close().
-	if err := w.writeCheckpointFile(); err != nil {
+	if err := w.writeCheckpointFile(
+		w.checkpointFilePath,
+		w.digestFdWithDigestContents.Digest().Sum32(),
+		w.digestBuf,
+	); err != nil {
 		w.err = err
 		return err
 	}
 	return nil
 }
 
+func (w *writer) DeferClose() (persist.DataCloser, error) {
+	err := w.close()
```

Comment:
Worth erroring if this is called on a closed writer?

Reply:
There's currently no mechanism tracking open/closed state and the upstream caller calls open/close in a single thread. I think it's fine to leave the expectations as is.